In [2]:
# default_exp molmap.gene.distance

# molmap.gene

> Gene feature processing using MolMap

# Add feature sums

The data we have in mind is a $(N, M)$ matrix, with $N$ being the number of genes and $M$ the number of features.

Here we expand the features by calculating the sums of each two features, this will result in a $(N, M^2)$ matrix. 

and split 

In [1]:
import umap
import numpy as np
import pandas as pd
from lapjv import lapjv
from sklearn.metrics import pairwise_distances
from scipy.spatial.distance import squareform
from scipy.cluster.hierarchy import fcluster, linkage
%config Completer.use_jedi = False

no need to untar the archive file

In [2]:
file_name = "/mnt/storage2/molmap_features/gene/input/All-TCGA-sample_log2expression-matrix.csv"

data_T = pd.read_csv(file_name, header = 0, index_col = 0)
data_T.shape

(16988, 17750)

we have 16,988 samples and 17,750 genes.

But what exactly are the data?

In [3]:
data_T.head()

TSPAN6      DPM1     SCYL3  C1orf112       FGR       CFH     FUCA2  \
ID-1 -4.299028  3.519240  1.536550  1.325214 -1.511691 -5.921390  0.749663   
ID-2  4.029771  4.813514  1.968828 -0.442026 -1.286304  2.061085  3.516658   
ID-3  3.116315  5.166434  1.999531  1.942496  3.857643  2.456412  3.869625   
ID-4  3.769412  4.538290  1.331419 -0.193965  2.272979  2.762285  4.021355   
ID-5       NaN  4.825831  2.337797  1.283092  6.389016 -1.501034  3.324018   

          GCLC      NFYA     STPG1  ...  DUS4L-BCAP29  AC016394.2  AL357079.3  \
ID-1  2.514021  5.151169 -0.718331  ...     -1.246053    2.770343    1.803310   
ID-2  1.823668  3.525380  1.390558  ...     -0.782893    1.574877    0.048376   
ID-3  2.112233  3.731694  0.115300  ...     -0.500014   -0.618606    0.765280   
ID-4  2.557386  2.982035  0.044044  ...     -0.400492    0.259182    0.364124   
ID-5  2.601649  3.896485 -0.235144  ...     -0.117318    2.482977    1.484860   

      AL357874.3   C8orf44  AL354833.1  AL133351.4  AC073611.1  AL592295.6  \
ID-1   -1.604279  0.465818   -0.824699   -1.683623   -1.761702    0.580338   
ID-2   -0.169096  0.550704   -5.304719   -1.763171   -2.417348    0.853038   
ID-3   -1.484388  1.604783   -3.172669   -0.844251   -1.522426    1.708540   
ID-4   -2.306859  0.780058   -4.097888    0.392867   -3.386846    2.698374   
ID-5   -1.158429  2.800144   -3.566613   -0.204566   -0.710520    3.071643   

      AP006621.6  
ID-1    0.477677  
ID-2   -0.276835  
ID-3   -0.412347  
ID-4   -2.871971  
ID-5   -0.962126  

[5 rows x 17750 columns]

In [12]:
data_T.index

Index(['ID-1', 'ID-2', 'ID-3', 'ID-4', 'ID-5', 'ID-6', 'ID-7', 'ID-8', 'ID-9',
       'ID-10',
       ...
       'ID-16979', 'ID-16980', 'ID-16981', 'ID-16982', 'ID-16983', 'ID-16984',
       'ID-16985', 'ID-16986', 'ID-16987', 'ID-16988'],
      dtype='object', length=16988)

In [13]:
data_T.columns

Index(['TSPAN6', 'DPM1', 'SCYL3', 'C1orf112', 'FGR', 'CFH', 'FUCA2', 'GCLC',
       'NFYA', 'STPG1',
       ...
       'DUS4L-BCAP29', 'AC016394.2', 'AL357079.3', 'AL357874.3', 'C8orf44',
       'AL354833.1', 'AL133351.4', 'AC073611.1', 'AL592295.6', 'AP006621.6'],
      dtype='object', length=17750)